### ISSUE TITLE
GUI redesign

### ISSUE THREAD

**Issue by jonorthwash** [SwatPhonLab/UltraTrace#93]
The GUI (at least the control area) needs to be redesigned a bit.

I like the results of our initial experiments with icons.  See also the concepts in `/dev`, e.g. https://github.com/SwatPhonLab/UltraTrace/blob/master/dev/gui-mockup.svg

**Comment by jonorthwash** [comment]
@keggsmurph21, I think this task would be up your alley (a little more than for the rest of us at least).

**Comment by keggsmurph21** [comment]
I'll look into this.  I had tried to get my hands dirty a few weeks ago, but got frustrated with `tkinter`, so I started writing a [declarative markup language that would wrap around tkinter](https://github.com/keggsmurph21/tkml) specifically for this purpose.  (In fact, I've been using [this](https://github.com/keggsmurph21/tkml/blob/master/examples/ultratrace.tkml) as my reference file.)

Finishing that in time for the conference might be a bit out of scope, but I can definitely take another stab at fixing the GUI.

**Comment by jonorthwash** [comment]
@keggsmurph21, any thoughts/progress on this?

**Comment by keggsmurph21** [comment]
Yeah I've thought a bit about it. @calebho and I think we could probably hack some improvements together with `tkinter`, but it would be relatively ugly / difficult to maintain.

We're going to explore some alternatives (probably do a small POC with some subset of `wxPython`, `PyQt`, and/or javascript via RPC) once we have the model API nailed down a bit more.

**Comment by jonorthwash** [comment]
I think some hacks in tkinter would be good for the abstract (due Jan 15).  Reimplementing everything can wait until after we have the abstract in.

**Comment by jonorthwash** [comment]
Hi @keggsmurph21, @calebho.  I know there's work on a UI rewrite in a branch.  Could someone fill me in on the vision guiding that?

I seem to remember some discussion about a browser-based front-end?  I ask now because I could see [blueprint.js's sidebar](https://dribbble.com/shots/3647198-Sidebar) (or similar) being a really useful addition for this project.

**Comment by kmurphy4** [comment]
Hey Jonathan -- we were thinking of writing an [Electron](https://www.electronjs.org/) frontend, which is very similar to running the application in a web browser.  We can take a look at the link you posted as the UI gets designed.  I've been swamped the past few weeks, but will hopefully have some time to work on this over the next few weeks.  Thanks for the heads up!
